src/brod_group_subscriber_v2.erl

```diff
@@ -0,0 +1,454 @@
+         , consumer_config => brod:consumer_config()
+         , group_config    => brod:group_config()
+         }.
+
+-type commit_fun() :: fun((brod:offset()) -> ok).
+
+-type init_info() ::
+        #{ group_id     := brod:group_id()
+         , topic        := brod:topic()
+         , partition    := brod:partition()
+         , commit_fun   := commit_fun()
+         }.
+
+-type member_id() :: brod:group_member_id().
+
+-callback init(brod_group_subscriber_v2:init_info(), _CbConfig) ->
+  {ok, _State}.
+
+-callback handle_message(brod:kafka_message(), State) ->
+      {ok, commit, State}
+    | {ok, ack, State}
+    | {ok, State}.
+
+%% Get committed offset (in case it is managed by the subscriber):
+-callback get_committed_offset(State) ->
+  {ok, brod:offset() | undefined, State}.
+
+-define(DOWN(Reason), {down, brod_utils:os_time_utc_str(), Reason}).
+
+-type worker() :: pid().
+
+-type topic_partition() :: {brod:topic(), brod:partition()}.
+
+-type workers() :: #{topic_partition() => workers()}.
+
+-type committed_offsets() :: #{topic_partition() =>
+                                 { brod:offset()
+                                 , boolean()
+                                 }}.
+
+-record(state,
+        { config                  :: subscriber_config()
+        , group_id                :: brod:group_id()
+        , coordinator             :: pid()
+        , generation_id           :: integer() | ?undef
+        , workers = #{}           :: workers()
+        , committed_offsets = #{} :: committed_offsets()
+        , cb_module               :: module()
+        , cb_config               :: term()
+        , client                  :: brod:client()
+        }).
+
+-type state() :: #state{}.
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+%% @doc Start (link) a group subscriber.
+%% Client:
+%%   Client ID (or pid, but not recommended) of the brod client.
+%% GroupId:
+%%   Consumer group ID which should be unique per kafka cluster
+%% Topics:
+%%   Predefined set of topic names to join the group.
+%%   NOTE: The group leader member will collect topics from all members and
+%%         assign all collected topic-partitions to members in the group.
+%%         i.e. members can join with arbitrary set of topics.
+%% GroupConfig:
+%%   For group coordinator, @see brod_group_coordinator:start_link/5
+%% ConsumerConfig:
+%%   For partition consumer, @see brod_consumer:start_link/4
+%% MessageType:
+%%   The type of message that is going to be handled by the callback
+%%   module. Can be either message or message set.
+%% CbModule:
+%%   Callback module which should have the callback functions
+%%   implemented for message processing.
+%% CbInitArg:
+%%   The term() that is going to be passed to CbModule:init/1 when
+%%   initializing the subscriger.
+%% @end
+-spec start_link(subscriber_config()) -> {ok, pid()} | {error, any()}.
```

Comment:
Maybe move these docs to the `subscriber_config` type, since they aren't parameters to the function.